tests/test_job.py

```diff
@@ -230,3 +231,36 @@ async def coro2():
 
     await fut
     await job.close()
+
+
+async def test_job_await_closed(scheduler) -> None:
+    async def coro() -> int:
+        return 5
+
+    job = await scheduler.spawn(coro())
+    assert not job._closed
+
+    # Let coro run.
+    await asyncio.sleep(0)
+    # Then let done callback run.
+    await asyncio.sleep(0)
+
+    assert job._closed
+    assert job._task.done()
+    assert await job.wait() == 5
+
+
+async def test_job_await_explicit_close(scheduler) -> None:
+    async def coro() -> None:
+        await asyncio.sleep(1)
+
+    job = await scheduler.spawn(coro())
+    assert not job._closed
+
+    await asyncio.sleep(0)
+    await job.close()
```

Comment:
```suggestion
    await job.close()
```

We are already waiting for the closing job here. Do we need to additionally yield back to the event loop?

Reply:
I wanted to ensure the task is started. If you close() before the task is started, then the task is cancelled without ever starting, which I think is a case covered elsewhere.